### ISSUE TITLE
nDPS and pDPS aren't in old reports, errors out on anything before endwalker when trying to get them

### ISSUE THREAD

**Issue by arieker** [halworsen/fflogsapi#4]
Key error is given when attempting to do this. Likely should make rdps/adps the default when doing calls on old logs instead of nDPS and/or warn the user when doing this on old stuff w/o those metrics.
ndps=rank['nDPS'],
KeyError: 'nDPS'

**Comment by halworsen** [comment]
Can you please post the full stack trace of the error and a link to the report for which the error occurs.

**Comment by arieker** [comment]
Updates/More info:

- Issue will occur even if manually overriding with rDPS, see sample code
- In this case at least, the error only occurs if the character actually has clears logged.
- Report does not matter, just pick something where nDPS is not a valid metric, so shadowbringers and earlier.
- Seems the temporary workaround is to just remove nDPS in character.py and dataclasses.py

Sample code:
```
from config import CLIENT_ID, CLIENT_SECRET
from fflogsapi import FFLogsClient, GQLEnum

client = FFLogsClient(CLIENT_ID, CLIENT_SECRET)
character = client.get_character(id=)

encounterRank = character.encounter_rankings(filters={
    'metric': GQLEnum('rdps'), #even when manually set to rDPS, still doesn't work
    'encounterID': 1047, #shb ucob, choose something character has a clear in
})

client.close()
client.save_cache()
```

Stacktrace:
```Traceback (most recent call last):
  File "C:\Users\User\source\repos\Dummy Repo\myfile.py", line 7, in <module>
    encounterRank = character.encounter_rankings(filters={
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\fflogsapi\characters\character.py", line 182, in encounter_rankings
    ndps=rank['nDPS'],
         ~~~~^^^^^^^^
KeyError: 'nDPS'

**Comment by halworsen** [comment]
Fixed in v2.1.1
